### PR TITLE
Upgrade query-string to v9

### PR DESCRIPTION
**Pull request by MohammedFaragallah**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ module.exports = {
         '/packages/create-react-admin/templates',
     ],
     transformIgnorePatterns: [
-        '[/\\\\]node_modules[/\\\\](?!(@hookform)/).+\\.(js|jsx|mjs|ts|tsx)$',
+        '[/\\\\]node_modules[/\\\\](?!(@hookform|query-string|decode-uri-component|filter-obj|split-on-first)/).+\\.(js|jsx|mjs|ts|tsx)$',
```
Comment (fzaninotto) [inline jest.config.js:31]: does that mean that react-admin users will have to do the same in their jest configuration?

Comment (djhi) [reply jest.config.js:31]: No, this is specific to Jest

**Comment by djhi** [comment]
Actually we are going to use `URLSearchParams` directly. Thanks!

**Comment by djhi** [comment]
Sorry for the noise. `query-string` actually does more that `URLSearchParams` and we decided to keep it. Can you rebase or merge the latest `next` branch? Thanks!

**Comment by MohammedFaragallah** [comment]
> Sorry for the noise. `query-string` actually does more that `URLSearchParams` and we decided to keep it. Can you rebase or merge the latest `next` branch? Thanks!

No worries happy to help any time, Done.

File: jest.config.js (hunk 1 above)
Comment (fzaninotto) [reply jest.config.js:31]: This is exactly my question

File: docs/DataProviderWriting.md [hunk 2]
```diff
@@ -687,7 +687,7 @@ Here is an example implementation, that you can use as a base for your own Data
 
 ```js
 import { fetchUtils } from 'react-admin';
-import { stringify } from 'query-string';
```
Comment (fzaninotto) [inline docs/DataProviderWriting.md:690]: Do we need to do that? I see no upgrade guide in query-string for that change.

Comment (zhu-hong) [reply docs/DataProviderWriting.md:690]: yeah, this package just export default now

**Comment by fzaninotto** [comment]
Thanks for your comment. Can you rebase to fix the merge conflicts?

**Comment by MohammedFaragallah** [comment]
> Thanks for your comment. Can you rebase to fix the merge conflicts?

Done but one of the checks is failing and there's no log

**Review by fzaninotto — APPROVED** [review]
This was a temporary error, I relaunched the job and it passed. Good job!

**Pull request merged**